src/utils.js

```diff
@@ -205,3 +205,22 @@ exports.withTimeout = (asyncFn, time) => {
     ])
   }
 }
+
+/**
+ * Iterates the given `asyncIterator` and runs each item through the given `asyncFn` in parallel.
+ * Returns a promise that resolves when all items of the `asyncIterator` have been passed
+ * through `asyncFn`.
+ *
+ * @param {AsyncIterable} [asyncIterator]
+ * @param {Function} [asyncFn]
+ * @returns {Array}
+ *
+ * @private
+ */
+exports.mapParallel = async function (asyncIterator, asyncFn) {
+  const tasks = []
+  for await (const item of asyncIterator) {
+    tasks.push(asyncFn(item))
+  }
+  return Promise.all(tasks)
```

Comment:
We have to wait for `asyncIterator` to end before parallel mapping over the items it yields. This is a little unusual and potentially buffers a large amount of data in memory.

Reply:
> We have to wait for asyncIterator to end before parallel mapping over the items it yields

i dont understand what you mean. that is neither a requirement we have nor the behavior of this code.

each item starts its `asyncFn` immediately when yielded. handling of each `asyncFn` does not block further iteration. In both my uses all I needed was to know when all have completed, so I returned a `Promise.all`

Reply:
ah you’re correct - my apologies I didn’t notice the function call!